netket/jax/_utils_random.py

```diff
@@ -30,16 +30,29 @@ def PRNGKey(
     The same seed will be distributed to all processes.
     """
     if seed is None:
-        key = jax.random.PRNGKey(random_seed())
-    elif isinstance(seed, int):
+        seed = random_seed()
+
+    if isinstance(seed, int):
```

Comment:
I'd rather not have this assumption. 
Because if someone is generating a seed using time.time or numpy the program is identical but the output is not.

The always stated purpose of `nkjax.PRNGKey` is to ensure a common seed everywhere regardless of what you do, and it costs us little to keep this guarantee.

Reply:
> I'm not a big fan of all of this babysitting, but ok. I would expect people knew what they are doing....

I'd like to start doing less babysitting, so I agree with you on this.

If you have a good argument for this case I'll listen! 
But nkjax.PRNGKey is really just there to ensure this...